### PR TITLE
add getMintlist

### PR DESCRIPTION
**Pull request by mertimus**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import axios, { AxiosError } from "axios";
-import { Webhook, CreateWebhookRequest, EditWebhookRequest } from "./types";
+import { Webhook, CreateWebhookRequest, EditWebhookRequest, CreateCollectionWebhookRequest, GetMintlistRequest, GetMintlistResponse, MintlistObject } from "./types";
 
 const API_URL_V0: string = "https://api.helius.xyz/v0";
-const API_URL_V1: string = "https://api.heliuys.xyz/v1";
```
Comment (vovkman) [inline src/index.ts:5]: nice

File: src/index.ts [hunk 2]
```diff
@@ -146,12 +146,92 @@ export class Helius {
...
+        if (firstVerifiedCreators != undefined && verifiedCollectionAddresses != undefined) {
+            throw new Error(`cannot provide both firstVerifiedCreators and verifiedCollectionAddresses. Please only provide one.`)
+        }
+
+        let mintlist: MintlistObject[] = [];
+        let query = {};
+
+        if (firstVerifiedCreators != undefined) {
+            query = { firstVerifiedCreators }
+        } else { // must have used verifiedCollectionAddresses
```
Comment (vovkman) [inline src/index.ts:171]: prob worth checking this anyways so we can return a sensical error if somehow happens

File: src/index.ts (hunk 1 above)
Comment (NicolasPennie) [reply src/index.ts:5]: real nice

File: src/index.ts [hunk 3]
```diff
@@ -146,12 +146,92 @@ export class Helius {
...
+
+        const { firstVerifiedCreators, verifiedCollectionAddresses } = request.collectionQuery;
+        if (firstVerifiedCreators != undefined && verifiedCollectionAddresses != undefined) {
+            throw new Error(`cannot provide both firstVerifiedCreators and verifiedCollectionAddresses. Please only provide one.`)
+        }
+
+        let mintlist: MintlistObject[] = [];
+        let query = {};
+
+        if (firstVerifiedCreators != undefined) {
```
Comment (NicolasPennie) [inline src/index.ts:169]: This check is unnecessary

File: src/types/index.ts [hunk 4]
```diff
@@ -4,12 +4,46 @@ export interface Webhook {
...
+    result: MintlistObject[],
+    paginationToken: string
+}
+
+export interface MintlistObject {
+    mint: string,
+    name: string,
+}
+
+export type CollectionIdentifier = {
```
Comment (NicolasPennie) [inline src/types/index.ts:32]: You can actually make this a one_of type.

export type FirstCreatorIdentifier = { firstVerifiedCreators }
export type VerifiedCollectionIdentifier = { verifiedCollectionAddresses }
export type NftCollectionIdentifier = FirstCreatorIdentifier | VerifiedCollectionIdentifier

**Pull request merged**